test/dtypes/test_nf4.py

```diff
@@ -57,42 +58,46 @@ def save_state_dict_to_buffer(self, state_dict: OrderedDict):
         buffer.seek(0)
         return buffer
 
-    def test_register_nf4_as_param(self):
-        nf4_tensor = to_nf4(torch.randn(512, 512, dtype=torch.bfloat16))
+    @pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float16, torch.float32])
+    def test_register_nf4_as_param(self, dtype: torch.dtype):
+        nf4_tensor = to_nf4(torch.randn(512, 512, dtype=dtype))
```

Comment:
do we need to remove bf16 in `to_nf4` as well?
```
def to_nf4(tensor, block_size: int = 64, scaler_block_size: int = 256):
    tensor1 = tensor.to(torch.bfloat16)
    return NF4Tensor.from_tensor(tensor1, block_size, scaler_block_size)
```

Reply:
to make torchtune fp32 work, I reuse this unit test directly: https://github.com/pytorch-labs/ao/pull/118

this one is passing now in my PR. there are still gaps in state dict and nn.functional

Reply:
great catch!